### PR TITLE
fix: small fixes

### PR DESCRIPTION
**Pull request by silverqx**
Small fixes discovered during reviews.

### CODE REVIEW

File: src/ccache/util/wincompat.hpp [hunk 1]
```diff
@@ -29,8 +29,8 @@
 // From:
 // http://mesos.apache.org/api/latest/c++/3rdparty_2stout_2include_2stout_2windows_8hpp_source.html
 #  ifdef _MSC_VER
-const mode_t S_IRUSR = mode_t{_S_IREAD};
-const mode_t S_IWUSR = mode_t{_S_IWRITE};
+constexpr auto S_IRUSR = mode_t{_S_IREAD};
+constexpr auto S_IWUSR = mode_t{_S_IWRITE};
```
Comment (silverqx) [inline src/ccache/util/wincompat.hpp:33]: No problem to revert it if you don't like it.

File: src/ccache/ccache.cpp [hunk 2]
```diff
@@ -2492,10 +2492,6 @@ do_cache_compilation(Context& ctx)
 
   ProcessArgsResult processed = process_args(ctx);
 
-  if (processed.error) {
-    return tl::unexpected(*processed.error);
-  }
-
```
Comment (silverqx) [inline src/ccache/ccache.cpp:2498]: This condition can never be false, it's useless? If you look [here](https://github.com/ccache/ccache/blob/master/src/ccache/argprocessing.cpp#L1747-L1752) and [here](https://github.com/ccache/ccache/blob/master/src/ccache/argprocessing.hpp#L34-L41).

**Review by silverqx — COMMENTED** [review]
I added comments to the modified code.

File: src/ccache/ccache.cpp (hunk 2 above)
Comment (jrosdahl) [reply src/ccache/ccache.cpp:2498]: You seem to have missed the https://github.com/ccache/ccache/blob/baea34b407f5f4f036fce8ce6f5d7873c572467f/src/ccache/argprocessing.hpp#L33 constructor, for instance invoked here: https://github.com/ccache/ccache/blob/baea34b407f5f4f036fce8ce6f5d7873c572467f/src/ccache/argprocessing.cpp#L1286

Comment (jrosdahl) [reply src/ccache/ccache.cpp:2498]: The construction comes from before `tl::expected` was introduced. I'll refactor it to use `tl::expected` instead.

Comment (silverqx) [reply src/ccache/ccache.cpp:2498]: I see there are 13 of them 😆 I totally missed it, the method is so long.

Comment (silverqx) [reply src/ccache/ccache.cpp]: I have reverted this change.

**Review by jrosdahl — APPROVED** [review]

File: src/ccache/ccache.cpp (hunk 2 above)
Comment (jrosdahl) [reply src/ccache/ccache.cpp]: > I'll refactor it to use tl::expected instead.

Done in 181ffcff3186.

**Pull request merged**